### PR TITLE
Velero installation guide and overrides docs

### PR DESCRIPTION
**Pull request by clebs**
**Description**

Changes proposed in this pull request:

- Add a user friendly installation guide for velero to the documentation.
- Add a ready to use overrides file to ease the installation.

**Related issue(s)**
#4835

### CODE REVIEW

File: docs/backup/03-01-install-velero.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  <summary>
+  Google Cloud Platform
+  </summary>
+  
+  For GCP, you need to have a bucket in `Google Cloud Storage`, and a service account able to access it and store data (and its JSON key).
+
+  1. Download the overrides file to your installation folder:
+      - [Velero Overrides](./assets/velero-overrides.yaml)
+
+  2. Run the following commands providing the necessary information to replace the placeholders on velero's configuration with your GCP bucket and credentials:
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
  2. To replace the placeholders on Velero's configuration with your GCP bucket and credentials, run:
```

File: docs/backup/03-01-install-velero.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          sed -i.bak "s/__PROVIDER__/$(echo -n gcp | base64)/g" velero-overrides.yaml
+
+          # Set the base64 encoded bucket name to the name of your bucket in GCS
+          sed -i.bak "s/__BUCKET__/$(echo -n <bucket name> | base64)/g" velero-overrides.yaml
+          
+          # Set the base64 encoded credentials JSON providing your service account key file
+          sed -i.bak "s/__CREDENTIALS__/$(base64 <credentials JSON file path>)/g" velero-overrides.yaml
+      ```
+
+  3. Run the kyma installation with the velero overrides:
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
  3. Run the Kyma installation with the Velero overrides:
```

File: docs/backup/03-01-install-velero.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      kyma install -o velero-overrides.yaml
+      ```
+      
+      </details>
+      <details>
+      <summary>
+      Cluster Installation
+      </summary>
+      
+      1. Install Kyma following the [installation guide](/root/kyma/#installation-installation)
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
      1. [Install](/root/kyma/#installation-installation) Kyma.
```
I usually add the link under "install" as it reduces the amount of information :)

File: docs/backup/03-01-install-velero.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      
+      2. Apply the overrides
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
      2. Deploy the `velero-overrides` file:
```
So here we apply the overrides, meaning we deploy the file on the cluster - make it available for the installer, right? If so, I would use `deploy` to not confuse the reader with two lines saying `apply the overrides` and then `use the installer to apply the overrides`. What do you think?

File: docs/backup/03-01-install-velero.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+          ```bash
+          kubectl apply -f velero-overrides.yaml
+          ```
+      3. Trigger the installer to apply the overrides
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
      3. Trigger the installer to apply the overrides:
```

File: docs/backup/03-01-install-velero.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+          ```bash
+          kubectl label installation/kyma-installation action=install
+          ```
+      
+      </details>
+      </div>
+
+  >**NOTE:** For more information on Kyma overrides visit the [Installation Overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation) section.
+
+  >**NOTE:** For more information on configuring and installing velero in GCP visit https://velero.io/docs/v1.0.0/gcp-config/
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
  >**NOTE:** For details on configuring and installing Velero in GCP, see [this](https://velero.io/docs/v1.0.0/gcp-config/) documentation.
```

File: docs/backup/03-01-install-velero.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  <details>
+  <summary>
+  Azure
+  </summary>
+
+  Coming soon...
+
+  >**NOTE:** For more information on Kyma overrides visit the [Installation Overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation) section.
+
+  >**NOTE:** For more information on configuring and installing velero in Azure visit https://velero.io/docs/v1.0.0/azure-config/
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]:
```suggestion
  >**NOTE:** For details on configuring and installing Velero in Azure, see [this] (https://velero.io/docs/v1.0.0/azure-config/) document.
```

File: docs/backup/03-01-install-velero.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  
+  </details>
+  <details>
+  <summary>
+  AWS
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]: If it's not supported, I would remove this tab. We can write in the overview section at the top that GCP and Azure are supported and AWS is not - then we don't need the tab.

File: docs/backup/03-01-install-velero.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
+---
+title: Install velero
+type: Details
+---
+In order to be able to backup and restore your Kyma cluster, [Velero](https://github.com/heptio/velero/) needs to be configured and installed.
+
+## Velero setup
+
+To sucessfully set up velero, you need to provide a supported storage location and credentials to access it. Please follow the instructions below for the provider of your choosing:
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/backup/03-01-install-velero.md]: Here we may also add a short note saying, for example:

`Currently, you can install Velero on GCP and Azure. AWS is not supported.` 
and get rid of the AWS tab.

File: docs/kyma/04-01-overview.md [hunk 10]
```diff
@@ -28,17 +28,20 @@ To make the local development process easier, we introduced the **Kyma Lite** co
 | `service-catalog-addons` | ✅ | ✅ |
 | `nats-streaming` | ✅ | ✅ |
 
+> \* In order to install velero, it needs to be set up first. Please follow the Kyma Backups setup instructions below.
```
Comment (bszwarc) [inline docs/kyma/04-01-overview.md]: I will confirm this part on Monday. 
(the best way to put it).

Comment (bszwarc) [inline docs/kyma/04-01-overview.md]:
```suggestion
> \* To install Velero, you need to set it up first. To do so, use the Kyma backup [setup instructions](/components/backup#overview-overview).
```

File: docs/kyma/04-01-overview.md [hunk 11]
```diff
@@ -28,17 +28,20 @@ To make the local development process easier, we introduced the **Kyma Lite** co
...
 - [Install Kyma locally](#installation-install-kyma-locally)
 - [Install Kyma on a cluster](#installation-install-kyma-on-a-cluster)
 
-Read rest of the installation documents to learn how to:
+Read the rest of the installation documents to learn how to:
 - [Disable the selected components' installation or install them separately](#configuration-custom-component-installation)
 - [Upgrade Kyma to a new version](#installation-upgrade-kyma)
 - [Update Kyma](#installation-update-kyma)
 - [Reinstall Kyma](#installation-reinstall-kyma)
+- [Setup Kyma Backups](/components/backup)
```
Comment (bszwarc) [inline docs/kyma/04-01-overview.md]:
```suggestion
- [Set up Kyma backup](/components/backup#overview-overview)
```

File: docs/backup/03-01-install-velero.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+  >**NOTE:** For more information on configuring and installing velero in GCP visit https://velero.io/docs/v1.0.0/gcp-config/
+
+  </details>
+  <details>
+  <summary>
+  Azure
+  </summary>
+
+  Coming soon...
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]: This part is still under construction?

File: docs/kyma/04-01-overview.md (hunk 11 above)
Comment (clebs) [reply docs/kyma/04-01-overview.md]: I do not think this is gramatically correct...
Either we say "Set up Kyma Backups" in plural so we do not need an article, or we say "Set up the kyma backup".
The current suggestion sounds strange to me .

File: docs/backup/03-01-install-velero.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+## Velero setup
+
+To sucessfully set up velero, you need to provide a supported storage location and credentials to access it. Please follow the instructions below for the provider of your choosing:
+  <div tabs>
+  <details>
+  <summary>
+  Google Cloud Platform
+  </summary>
+  
+  For GCP, you need to have a bucket in `Google Cloud Storage`, and a service account able to access it and store data (and its JSON key).
```
[the remark this answers is not in the document]
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: invalid

File: docs/backup/03-01-install-velero.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+  1. Download the overrides file to your installation folder:
```
[the remark this answers is not in the document]
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: invalid

File: docs/backup/03-01-install-velero.md (hunk 12 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Yes, I just got an Azure subscription recentlyand need to run some tests before providing a working configuration example.

Anyway, the Azure specific parameters are listed in the configuration section and can be used already.

File: docs/backup/03-01-install-velero.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+      - [Velero Overrides](./assets/velero-overrides.yaml)
```
[the remark this answers is not in the document]
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Invalid.

File: docs/backup/03-01-install-velero.md (hunk 1 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Invalid.

File: docs/backup/03-01-install-velero.md (hunk 5 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Invalid.

File: docs/backup/03-01-install-velero.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
+---
+title: Install velero
```
[the remark this answers is not in the document]
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+type: Details
+---
+In order to be able to backup and restore your Kyma cluster, [Velero](https://github.com/heptio/velero/) needs to be configured and installed.
```
[the remark this answers is not in the document]
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 2 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 3 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 4 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 6 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 7 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md (hunk 8 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Fixed thanks!

File: docs/kyma/04-01-overview.md (hunk 11 above)
Comment (clebs) [reply docs/kyma/04-01-overview.md]: Fixed thanks!

File: docs/kyma/04-01-overview.md (hunk 10 above)
Comment (clebs) [reply docs/kyma/04-01-overview.md]: Fixed thanks!

File: docs/backup/03-01-install-velero.md [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+                    }
+    ```
+    >**NOTE:** For details on configuring and installing Velero in GCP,  see [this](https://velero.io/docs/v1.0.0/gcp-config/) document.
+    </details>
+    <details>
+    <summary>
+    Azure
+    </summary>
+
+    Coming soon...
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]: When? :D Just a question - will we have something here for the release?

File: docs/backup/03-01-install-velero.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+
+    >**NOTE:** For details on configuring and installing Velero in Azure,  see [this](https://velero.io/docs/v1.0.0/azure-config/) document.
+    
+    </details>
```
Comment (bszwarc) [inline docs/backup/03-01-install-velero.md]: Will there be something here? If not, maybe we can combine two notes?

File: docs/backup/03-01-install-velero.md (hunk 18 above)
Comment (clebs) [reply docs/backup/03-01-install-velero.md]: Yes, I will work on this right after finishing this docs.

**Pull request merged**